techniques/system/common/1.0/rudder-stdlib-core.st

```diff
@@ -122,8 +122,28 @@ bundle agent rudder_common_report(technique_name, status, identifier, component_
 # index          : the current index, that will distinguish two identical reports and ensure they are both sent
 bundle agent rudder_common_report_index(technique_name, status, identifier, component_name, component_key, message, index)
 {
+  vars:
+    !(dry_run|global_dry_run)::
+      "new_status"  string => "${status}";
+    (dry_run|global_dry_run)::
+      # map result needed for direct calls to rudder_common_report
+      # A quick remark: the mapping result_error > audit_noncompliant is not complete, as the case n errors + 1 repaired leads to result_error in enforce but it leads to audit_error in audit (error wins in enforce, but repair wins in audit)
+      "new_status"  string => ifelse(strcmp("${status}", "result_na"),       "audit_na",           "${status}");
+      "new_status"  string => ifelse(strcmp("${status}", "result_success"),  "audit_compliant",    "${status}");
+      "new_status"  string => ifelse(strcmp("${status}", "result_repaired"), "audit_error",        "${status}");
+      "new_status"  string => ifelse(strcmp("${status}", "result_error"),    "audit_noncompliant", "${status}");
```

Comment:
why not using defaults ?
https://docs.cfengine.com/lts/reference-promise-types-defaults.html